File: Java/V1SDKExamples/src/com/versionone/sdk/examples/SSOExample.java
package com.versionone.sdk.examples;

import java.util.Calendar;
import java.util.Date;

import com.versionone.apiclient.APIException;
import com.versionone.apiclient.ConnectionException;
import com.versionone.apiclient.IAPIConnector;
import com.versionone.apiclient.IAssetType;
import com.versionone.apiclient.IAttributeDefinition;
import com.versionone.apiclient.IMetaModel;
import com.versionone.apiclient.IServices;
import com.versionone.apiclient.MetaModel;
import com.versionone.apiclient.OidException;
import com.versionone.apiclient.Query;
import com.versionone.apiclient.QueryResult;
import com.versionone.apiclient.Services;
import com.versionone.sso.V1SsoConnector;

public class SSOExample implements ISdkSample {

	public void loadConfig() {}
	
	public void commandLineArguments(String[] args) {}
	
	public void connect() {
    	        
        try {
            Log("Creating Meta Connector");
            IAPIConnector metaConnector = V1SsoConnector.createMetaConnector(Configuration.instance());
            
            Log("Create MetaModel");
            _meta = new MetaModel(metaConnector);
        	
            Log("Creating Data Connector");
            V1SsoConnector dataConnector = V1SsoConnector.createDataConnector(Configuration.instance(), false);
        	
        	Log("Authenticate");
			dataConnector.authenticate();
			
	        Log("Create Services");
	        _services = new Services(_meta, dataConnector);
			
		} catch (SecurityException e) {
			e.printStackTrace();
			return;
		} catch (ConnectionException e) {
			e.printStackTrace();
			return;
		}
        
	}

    public void run()
    {
    	try {
    		Log("Get Members");

            IAssetType memberType = getMetaModel().getAssetType("Member");
            IAttributeDefinition memberName = memberType.getAttributeDefinition("Name");

            Query query = new Query(memberType);
            query.getSelection().add(memberName);
            QueryResult result = getServices().retrieve(query);
            Log(String.format("Server returned %s members", result.getTotalAvaliable()));
    					
			Log("The End - Success");
		} catch (SecurityException e) {
			e.printStackTrace();
		} catch (ConnectionException e) {
			e.printStackTrace();
		} catch (APIException e) {
			e.printStackTrace();
		} catch (OidException e) {
			e.printStackTrace();
		} catch (Exception e) {
			e.printStackTrace();
		}    	
    }

    protected IMetaModel _meta; 
	public IMetaModel getMetaModel() {
		return _meta;
	}

	private IServices _services;
	public IServices getServices() {
		return _services;
	}
        
	private void Log(String message)
    {
        Date now = Calendar.getInstance().getTime();
        System.out.printf("%s - %s\n", now.toString(), message);
    }
    
}

File: Net/SSOExtension/V1SSOExample/SsoExample.cs
using System;
using VersionOne.SDK.APIClient;
using VersionOne.SDK.APIClient.SSOExtension;

namespace V1SSOExample
{
    /// <summary>
    /// Demonstrate SSO connector
    /// </summary>
    internal class SsoExample : ISdkSample
    {
        public IMetaModel Meta { get; private set; }
        public IServices Services { get; private set; }

        public void CommandLineArguments(string[] args) {}
        public void LoadConfig() {}

        public void Connect()
        {
            Log("Create Meta Connector");
            var metaConnector = V1SsoConnector.CreateMetaConnection(Configuration.Instance());

            Log("Created Meta Model");
            Meta = new MetaModel(metaConnector);

            Log("Creating Data Connector");
            var dataConnector = V1SsoConnector.CreateDataConnection(Configuration.Instance(), false); 

            Log("Authenticate");
            dataConnector.Authenticate();

            Log("Created Services");
            Services = new Services(Meta, dataConnector);
        }

        public void Run()
        {
            Log("Get some meta stuff");
            IAssetType memberType = Meta.GetAssetType("Member");
            Log(memberType.DisplayName);

            Log("Perform a query");
            var query = new Query(memberType);
            QueryResult result = Services.Retrieve(query);

            Log(string.Format("Server returned {0} members", result.TotalAvaliable));
        }

        private static void Log(string message)
        {
            var now = DateTime.Now;
            Console.WriteLine("{0} {1} - {2}", now.ToShortDateString(), now.ToLongTimeString(), message);
        }

    }
}